### ISSUE TITLE
#abs/2009jinfo...3....1h and #abs/2009JInfo...3....1H

### ISSUE THREAD

**Issue by romanchyla** [adsabs/bumblebee#345]
the first one is empty, the second one works - but they should work both, or not?

**Comment by aaccomazzi** [comment]
Here we get into murky business, which brings us back to the issue of minting and using URIs for our resources.  Because URIs are case-insensitive, the following two URLs work as expected in classic, as they should:

http://adsabs.harvard.edu/abs/2009jinfo...3....1h
http://adsabs.harvard.edu/abs/2009JInfo...3....1H

We need to maintain this level of consistency with BB, so that people can predictably link to our pages.  But the links I'm envisioning are the same as described above (or very similar ones such as http://ui.adsabs.harvard.edu/abs/2009JInfo...3....1H), which would then redirect users to BB-friendly URLs for all users except for robots and/or applications.  I'm not sure how we do this, but there must be a way to detect when the client is javascript capable and use that information to act on the redirect policy.

So to sum up, I think we need:
1. an endpoint which acts as a redirector for URLs on our website, managing all resources under its top-level base (say http://adsabs.harvard.edu/abs)
2. the service looks up bibcodes in a case-insensitive way, resolves them to their canonical form, and then issue the proper redirects to BB pages or robot-friendly pages
3. the service is also aware of other identifiers as we've done in ADS classic, so that arXiv ids and/or dois can be resolved as well
4. we should find a way to encourage or force people to use these canonical uris for bookmarking, linking, but I'm not sure how to do that given that BB will display its #/abs/bibcode equivalent (and people will simply cut/paste from the location bar in their browser)

**Comment by romanchyla** [comment]
as we discovered today, when an alternate_bibcode is used, it fails to find the record

we should do redirection

**Comment by romanchyla** [comment]
solved in bumblebee; both case sensitive issues and alternate bibcodes + doi (ie. one could use any identifier in the url; bbb will resolve and redirect to the canonical bibcode)

however, i would not recommend to expose non-canonical links (ie. we should not advertise them)

and also, there is a confusion in your argumentation: we cannot use a service in /abs - that is where the client lives; unless you advocate to run a service which decides what to send...
